listenbrainz/webserver/static/js/src/listens/ListenCard.tsx

```diff
@@ -0,0 +1,207 @@
+    this.APIService = new APIService(
+      props.apiUrl || `${window.location.origin}/1`
+    );
+  }
+
+  componentDidMount(): void {
+    this.fetchCoverArt();
+  }
+
+  fetchCoverArt = async () => {
+    const { listen } = this.props;
+    const releaseMBID = _get(
+      listen,
+      "track_metadata.additional_info.release_mbid"
+    );
+    const recordingMSID = _get(
+      listen,
+      "track_metadata.additional_info.recording_msid"
+    );
+
+    this.APIService.getCoverArt(releaseMBID, recordingMSID).then((imageUrl) => {
+      const coverArtUrl = imageUrl !== "" ? "imageUrl" : DEFAULT_COVER_ART_URL;
+      this.setState({ coverArtUrl });
+    });
+  };
+
+  render() {
+    const { listen, mode, className } = this.props;
+    const { feedback, coverArtUrl } = this.state;
+
+    return (
+      <Card
+        onDoubleClick={this.playListen}
+        className={`listen-card row ${className}`}
+        title="Double click to play"
+      >
+        <MediaQuery minWidth={768}>
+          <div className="col-xs-10">
+            <div className="col-xs-9">
+              <div
+                className="cover-art img-responsive"
+                style={{
+                  background: `url("${coverArtUrl}")`,
+                }}
+              />
+              <div className="track-details">
+                <p title={listen.track_metadata.track_name}>
+                  {getTrackLink(listen)}
+                </p>
+                <p>
+                  <small
+                    className="text-muted"
+                    title={listen.track_metadata.artist_name}
+                  >
+                    {getArtistLink(listen)}
+                  </small>
+                </p>
+              </div>
+            </div>
+            <div className="col-xs-3">
+              {listen.playing_now ? (
+                <span className="listen-time text-center text-muted">
+                  <FontAwesomeIcon icon={faMusic as IconProp} /> Playing now
+                </span>
+              ) : (
+                <span
+                  className="listen-time text-center text-muted"
+                  title={
+                    listen.listened_at_iso?.toString() ||
+                    new Date(listen.listened_at * 1000).toISOString()
+                  }
+                >
+                  {listen.listened_at_iso
+                    ? timeago.ago(listen.listened_at_iso)
+                    : timeago.ago(listen.listened_at * 1000)}
+                </span>
+              )}
+            </div>
+          </div>
+
+          <div className="col-xs-2 text-center">
+            {mode === "follow" || mode === "recent" ? (
+              <a
+                href={`/user/${listen.user_name}`}
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                {listen.user_name}
+              </a>
+            ) : (
+              <div>
+                <FontAwesomeIcon icon={faHeart as IconProp} />
```

Comment:
These buttons won't do anything for now, right? If so, we should remove them and add them in a subsequent PR

Reply:
Okay will do